Tahi-project/tahi
# APERTA-10364 User can be assigned to a task

JIRA issue: https://jira.plos.org/jira/browse/APERTA-10364

#### What this PR does:
It allows assigning an user to a card, only roles who were set the action 'Can be assigned' in the 'Permissions' tab will be listed in the 'Assign' widget, also this widget will only be rendered to the roles who were set the action 'Can assign others'.
- A new migration was added, we now store the assigned_user_id in the tasks table, also the user is serialized as part of the task json response.
- A new endpoint was added to filter out users who can be assigned to a task
- overlay-task and participant-selector components have most of the changes.

#### Special instructions for Review or PO:
This is what I did to test this feature:
- I found an user who had the role 'Handling Editor' and in the card config 'Permissions' tab checked the action 'Can assign others'
- For the role 'Reviewer' I checked the action 'Can be assigned'
- Publish the card and add it to any of the workflow columns
- Log in as the user who has the role 'Handling Editor' and open up that card, in there you should see the 'Assign: +' widget. Assign and remove users, only users **who can be assigned** should be listed in there.
- Log in as any other user who can not assign users to a card but can go to the workflow and open up the card, make sure he/she doesn't see the ``assign`` widget.


#### Major UI changes
![assignuserwidget](https://user-images.githubusercontent.com/30911/30566249-e49a9b9c-9c90-11e7-8cad-8aa28493e578.png)

---

#### Code Review Tasks:

**Author tasks** (delete tasks that don't apply to your PR, this list should be finished before code review):

- [x] If I changed the database schema, I enforced database constraints.
- [x] I have ensured that the Heroku Review App has successfully deployed and is ready for PO UAT.

**Reviewer tasks** (these should be checked or somehow noted before passing on to PO):
